src/mainwindow.cpp

```diff
@@ -2817,34 +2827,21 @@ void MainWindow::highlightSearch() const
 {
     QString searchString = m_searchEdit->text();
 
-    if(!searchString.isEmpty()){
-        m_textEdit->blockSignals(true);
-
-        QTextDocument *document = m_textEdit->document();
-
-        QTextCursor highlightCursor(document);
-        QTextCursor cursor(document);
-
-        cursor.beginEditBlock();
-
-        QTextCharFormat colorFormat(highlightCursor.charFormat());
-        colorFormat.setBackground(Qt::yellow);
-
-        QTextCursor pos = document->find(searchString, 0);
-        if (!pos.isNull())
-            m_textEdit->setTextCursor(pos);
-
-        while (!highlightCursor.isNull() && !highlightCursor.atEnd()) {
-            highlightCursor = document->find(searchString, highlightCursor);
-
-            if (!highlightCursor.isNull()) {
+    if(searchString.isEmpty())
+        return;
 
-                highlightCursor.mergeCharFormat(colorFormat);
-            }
-        }
+    m_textEdit->moveCursor(QTextCursor::Start);
 
-        cursor.endEditBlock();
+    QList<QTextEdit::ExtraSelection> extraSelections;
+    while(m_textEdit->find(searchString)) {
+        QTextEdit::ExtraSelection extra = QTextEdit::ExtraSelection();
+        extra.format.setBackground(Qt::yellow);
+        extra.cursor = m_textEdit->textCursor();
+        extraSelections.append(extra);
```

Comment:
The format is always the same, so we can share it:

```c++
QTextCharFormat highlightFormat;
highlightFormat.setBackground(Qt::yellow);

QList<QTextEdit::ExtraSelection> extraSelections;
while(m_textEdit->find(searchString))
    extraSelections.append({ highlightFormat, m_textEdit->textCursor() });

Reply:
Done